### PR TITLE
fix(asynccomponent): Get location from props if no router context

### PR DESCRIPTION
**Pull request by lynnagara**
Router might not always be in context wherever the AsyncComponent is used.

Get the location from props if that's the case

### CODE REVIEW

**Comment by mattrobenolt** [comment]
Does this also fix https://sentry.io/sentry/javascript/issues/436072596/ ?

It seems like it does.

**Comment by lynnagara** [comment]
Yep. Should fix https://sentry.io/sentry/javascript/issues/436072596/, https://sentry.io/sentry/javascript/issues/463722757/ https://sentry.io/sentry/javascript/issues/463706261.

File: src/sentry/static/sentry/app/components/asyncComponent.jsx [hunk 1]
```diff
@@ -30,10 +30,19 @@ class AsyncComponent extends React.Component {
...
 
   componentWillReceiveProps(nextProps, nextContext) {
+    const isRouterInContext = !!this.context.router;
+
+    const currentLocation = isRouterInContext
+      ? this.context.router.location
+      : this.props.location;
+    const nextLocation = isRouterInContext
+      ? nextContext.router.location
+      : this.context.router.location;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/asyncComponent.jsx]: shouldn't this be props?

File: src/sentry/static/sentry/app/components/asyncComponent.jsx [hunk 2]
```diff
@@ -30,10 +30,19 @@ class AsyncComponent extends React.Component {
   }
 
   componentWillReceiveProps(nextProps, nextContext) {
+    const isRouterInContext = !!this.context.router;
+
+    const currentLocation = isRouterInContext
+      ? this.context.router.location
+      : this.props.location;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/asyncComponent.jsx]: I think `router` can also be a prop, we should just check that too

File: src/sentry/static/sentry/app/components/asyncComponent.jsx (hunk 1 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/components/asyncComponent.jsx]: Agh, thanks

File: src/sentry/static/sentry/app/components/asyncComponent.jsx (hunk 2 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/components/asyncComponent.jsx]: Really? We've never supported that though.

Comment (billyvg) [reply src/sentry/static/sentry/app/components/asyncComponent.jsx]: Looks like location is always provided so this should be fine

**Review by billyvg — APPROVED** [review]

**Pull request merged**